lib/urldata.h

```diff
@@ -327,8 +327,10 @@ struct kerberos5data {
 #if defined(USE_NTLM)
 struct ntlmdata {
   curlntlm state;
-#ifdef USE_WINDOWS_SSPI
+#if defined(USE_WINDOWS_SSPI)
+#if defined(_MSC_VER) && (_MSC_VER >= 1600)
```

Comment:
Can you add a comment explaining why this check is here and necessary?

Reply:
I was just about to comment here too - you probably want something different that the compiler version here, don't you?

Reply:
The SecPkgContext_Bindings requires Windows 7 or later. Visual Studio 2008 on appveyor fails to compile as it uses an older Windows SDK and where the structure is not available.
https://docs.microsoft.com/en-us/windows/desktop/api/sspi/ns-sspi-_secpkgcontext_bindings

For example at line 278 in the logs of https://ci.appveyor.com/project/curlorg/curl/builds/20617060/job/qmog8mlh34mnl3hc shows the compilation error.

Is there a better way to pass the compilation stage on VS2008?

Reply:
I've added a comment explaining the above https://github.com/curl/curl/pull/3321/files#diff-f43d2eae345916ac30849058f282a3f3R331. Do you think that's enough?

Reply:
Sorry, I missed your comment. The Windows 7 SDK can be used with Visual Studio 2005 and later if you want to, so I'd rather check the target Windows version.

If I read that correctly, the offending definitions (`SecPkgContext_Bindings` and `SECPKG_ATTR_ENDPOINT_BINDINGS`) are not dependent on the target Windows version. In that case, you could maybe use `WIN32_WINNT_MAXVER` from winsdkver.h. I can't remember if it's included by default, though. If not, it will be more complicated as IIRC classic MinGW and older Windows SDK versions don't have that header. There are also some definitions for the Windows SDK version in ntverp.h, but I haven't ever used these and I also cannot find that header in the MinGW folder.

Reply:
OK, I finally got my head around this. @MarcelRaad you are correct, we need to check the SDK version and not the compiler version. I've got two solution to propose here:
The first would be include the winsdkver.h and check _WIN32_WINNT_MAXVER
```
- #if defined(_MSC_VER) && (_MSC_VER >= 1600)
+ #include "winsdkver.h"
+ #if defined(_WIN32_WINNT_MAXVER) && defined(_WIN32_WINNT_WIN7) && (_WIN32_WINNT_MAXVER >= _WIN32_WINNT_WIN7)
```
or we just check the if _WIN32_WINNT_WIN7 is defined. It is defined on the Win7 SDK and all the later SKDs
```
- #if defined(_MSC_VER) && (_MSC_VER >= 1600)
+ #ifdef _WIN32_WINNT_WIN7
```
Which one you think we should choose?

Reply:
I like the second one, great idea! It's simpler as winsdkver.h isn't always available.
Do you happen to have classic MinGW or MinGW-w64 installed? I guess some needed definitions will probably be missing at least in classic MinGW. If you don't, I can test locally and push a fix-up commit together with this.

Reply:
@MarcelRaad Any news with this? Do you need any help by me?

Reply:
Ah, sorry, I was waiting for you to push your changes.

Please commit the `WIN32_WINNT_WIN7` change, rebase onto latest master and squash all your commits into the first one. Its commit message is already great, just one very minor thing: I think we don't use square brackets, it should be "ntlm_sspi:" instead of "[ntlm]".

Then I will test on MinGW and eventually push it.